cmd/katapult/output.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+var outputFlag, templateFlag string
+
+// Output is used to define the interface of outputs.
+type Output interface {
+	// JSON is used to return a string of the JSON output.
+	JSON() (json.RawMessage, error)
+
+	// YAML is used to return a string of the YAML output.
+	YAML() (string, error)
+
+	// Text is used to render a template. If string is blank, uses the default.
+	Text(template string) (string, error)
+}
+
+// Used to implement Output for a variety of use cases.
+type genericOutput struct {
+	item interface{}
+	tpl  string
+}
+
+// JSON is used to return a string of the JSON output.
+func (g genericOutput) JSON() (json.RawMessage, error) {
```

Comment:
I think we should probably change all the methods on `genericOutput` to use pointer receivers to avoid needless duplication of its memory.

```suggestion
func (g *genericOutput) JSON() (json.RawMessage, error) {
```